Kgaut chandima file edits a1

kgauthreaux opened the pull request.

hewanadun reviewed (commented):
@kgauthreaux 
CSV file format looks good. 
I have few questions/comments,

1. Are you still working on header variables (FRMDATEA1, LANGA1, etc.) error checks? (could not find those in the error checks CSV).
2. For date format validations, include both mm/dd/yyyy and yyyy/mm/dd as acceptable formats
3. Just to clarify, if variable X is compared with variables Y, Z - comp_vars column will have all 3 variable names (X, Y, Z), is that correct?
4. Similarly, if form A is compared with forms B, C - comp_forms column will have all 3 form names (A, B, C)?
5. I think we can fill the comp_forms column only if multiple forms included in the check (current plausibility checks has comp_forms column filled for within form checks).

kgauthreaux commented:
Thanks @hewanadun I made edits to the A1 files based on your comments! Answers below:

- Are you still working on header variables (FRMDATEA1, LANGA1, etc.) error checks? (could not find those in the error checks CSV).
_- Yes, you and I talked about this, but I am going to make sure we add these checks to the end of the list so that we just add new error codes instead of them changing later_
- For date format validations, include both mm/dd/yyyy and yyyy/mm/dd as acceptable formats
- _Okay, I added this to the questions_and_vars list and will keep in mind for error checks_
- Just to clarify, if variable X is compared with variables Y, Z - comp_vars column will have all 3 variable names (X, Y, Z), is that correct?
- _Correct_
- Similarly, if form A is compared with forms B, C - comp_forms column will have all 3 form names (A, B, C)?
- _Correct_
- I think we can fill the comp_forms column only if multiple forms included in the check (current plausibility checks has comp_forms column filled for within form checks).
- _Okay, sounds good, I removed 'a1' from that column for the current a1 plausibility checks_

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+error_code,error_no,error_type,form_name,packet,var_name,check_type,test_name,short_desc,full_desc,test_logic,comp_forms,comp_vars,do_in_redcap,in_prev_versions,questions
+a1-ivp-p-1001,1001,Alert,a1,I,RACEWHITE,Plausibility,"If one race is known, cannot check Unknown","If  RACEWHITE, RACEBLACK, RACEAIAN, RACENHPI, RACEASIAN, ETHISPANIC, or RACEOTH=1 then RACEUNKN cannot equal 1","If  Q2a. RACEWHITE, Q2b. RACEBLACK, Q2c. RACEAIAN, Q2d. RACENHPI, Q2e. RACEASIAN, Q2f. ETHISPANIC, or Q2g. RACEOTH=1 then Q2h. RACEUNKN cannot equal 1",IF (RACEWHITE=1 or RACEBLACK=1 or R ...
```
Please change the var_name for error a1-ivp-p-1001 to "RACEUNKN", it's easy to implement this check under that variable.

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+ETHISPANIC,
+RACEOTH,
+RACEUNKN",No,No,
+a1-ivp-p-1002,1002,Alert,a1,I,GENMAN,Plausibility,"If one gender is known, cannot check Don't know","If  GENMAN, GENWOMAN, GENTRMAN, GENTRWOMAN, GENNONBI, GENTWOSPIR, GENOTH, or GENNOANS=1 then GENDKN cannot equal 1","If  Q3a. GENMAN, Q3b. GENWOMAN, Q3c. GENTRMAN, Q3d. GENTRWOMAN, Q3e. GENNONBI, Q3f. GENTWOSPIR, Q3g. GENOTH, or Q3i. GENNOANS=1 then Q3hg. GENDKN cannot equal 1",IF (GENMAN=1 or GEN ...
```
Please change the var_name for error a1-ivp-p-1002 to "GENDKN", it's easy to implement this check under that variable.

kgauthreaux replied:
GENDKN or GENNOANS (what I'm changing error a1-ivp-p-1003 to)

hewanadun replied:
a1-ivp-p-1002 to "GENDKN" and a1-ivp-p-1003 to "GENNOANS"

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+GENOTH,
+GENDKN,
+GENNOANS",No,No,
+a1-ivp-p-1003,1003,Alert,a1,I,GENMAN,Plausibility,"If one gender is known, cannot check No answer","If  GENMAN, GENWOMAN, GENTRMAN, GENTRWOMAN, GENNONBI, GENTWOSPIR, GENOTH, or GENDKN=1 then GENNOANS cannot equal 1","If  Q3a. GENMAN, Q3b. GENWOMAN, Q3c. GENTRMAN, Q3d. GENTRWOMAN, Q3e. GENNONBI, Q3f. GENTWOSPIR, Q3g. GENOTH, or Q3h. GENDKN=1 then Q3i. GENNOANS cannot equal 1",IF (GENMAN=1 or GENWO ...
```
Please change the var_name for error a1-ivp-p-1003 to "GENNOANS", it's easy to implement this check under that variable.

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+GENOTH,
+GENDKN,
+GENNOANS",No,No,
+a1-ivp-p-1004,1004,Alert,a1,I,SEXORNGAY,Plausibility,"If one sexual orientation is known, cannot check Don't know","If  SEXORNGAY, SEXORNHET, SEXORNBI, SEXORNTWOS, SEXORNOTH, or SEXORNNOAN=1 then SEXORNDNK cannot equal 1","If Q6a. SEXORNGAY, Q6b. SEXORNHET, Q6c. SEXORNBI, Q6d. SEXORNTWOS, Q6e. SEXORNOTH, or Q6g. SEXORNNOAN=1 then Q6f. SEXORNDNK cannot equal 1",IF (SEXORNGAY=1 or SEXORNHET=1 or SE ...
```
Please change the var_name for error a1-ivp-p-1004 to "SEXORNDNK", it's easy to implement this check under that variable.

kgauthreaux replied:
Changing it but just want to double check that it's not supposed to be SEXORNNOAN like below

hewanadun replied:
a1-ivp-p-1004 to "SEXORNDNK" and a1-ivp-p-1005 to "SEXORNNOAN" (not the same)

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+SEXORNOTH,
+SEXORNDNK,
+SEXORNNOAN",No,No,
+a1-ivp-p-1005,1005,Alert,a1,I,SEXORNGAY,Plausibility,"If one sexual orientation is known, cannot check No answer","If  SEXORNGAY, SEXORNHET, SEXORNBI, SEXORNTWOS, SEXORNOTH, or SEXORNDNK=1 then SEXORNNOAN cannot equal 1","If Q6a. SEXORNGAY, Q6b. SEXORNHET, Q6c. SEXORNBI, Q6d. SEXORNTWOS, Q6e. SEXORNOTH, or Q6f. SEXORNDNK=1 then Q6g. SEXORNNOAN cannot equal 1",IF (SEXORNGAY=1 or SEXORNHET=1 or SEX ...
```
Please change the var_name for error a1-ivp-p-1005 to "SEXORNNOAN", it's easy to implement this check under that variable.

hewanadun commented on forms/uds/a1/form_a1_ivp_error_checks_p.csv:
```diff
@@ -0,0 +1,52 @@
+SEXORNOTH,
+SEXORNDNK,
+SEXORNNOAN",No,No,
+a1-ivp-p-1006,1006,Alert,a1,I,EXPANCEST,Plausibility,"If one main reason for the experience is known, cannot check No answer","If  EXPANCEST, EXPGENDER, EXPRACE, EXPAGE, EXPRELIG, EXPHEIGHT, EXPWEIGHT, EXPAPPEAR, EXPSEXORN, EXPEDUCINC, EXPDISAB, or EXPSKIN=1 then EXPNOANS cannot equal 1","If Q22a. EXPANCEST, Q22b. EXPGENDER, Q22c. EXPRACE, Q22d. EXPAGE, Q22e. EXPRELIG, Q22f. EXPHEIGHT, Q22g. EXPW ...
```
Is this a check for A1 form, can't find these variables in questions_and_vars sheet? And Q22 in A1 is a different question than what is mentioned here.

kgauthreaux replied:
Oops, yes these were originally in A1 and got moved to A1a!

hewanadun reviewed (changes requested):
@kgauthreaux Please check my comments on the files. Thanks!

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-c-059,059,Error,a1,I,ETHSCOTT,Conformity,ETHSCOTT conforms,ETHSCOTT must be blank or 1,Q3g6. ETHSCOTT (ethnic Scottish) must be blank or 1,IF ETHSCOTT is not blank or ETHSCOTT is not 1,,,Yes,No,
+a1-ivp-c-060,060,Error,a1,I,ETHWHIOTH,Conformity,ETHWHIOTH conforms,ETHWHIOTH must be blank or 1,Q3g7. ETHWHIOTH (other white ethnicity) must be blank or 1,IF ETHWHIOTH is not blank or ETHWHIOTH is not 1,,,Yes,No,
+a1-ivp-m-061,061,Error,a1,I,ETHWHIOTHX,Missingness,ETHWHIOTHX conditionally blank,If ETHWHIOTH is blank then ETHWHIOTHX must be blank,If Q3g7. ETHWHIOTH (other white ethnicity) is blank then Q3g7a. ETHWHIOTHX (other white ethnicity specify) must be blank,IF ETHWHIOTH is blank and ETHWHIOTHX is not blank,,,Yes,No,
+a1-ivp-c-062,062,Error,a1,I,RACEUNKN,Conformity,RACEUNKN conforms,RACEUNKN must be blank or 1,Q3h. RACEUNKN (race unknown) must be blank or 1,"IF RACEUNKN is not 0,1",,,Yes,No,
```
Yes, fixing in the file for re-upload!

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-m-100,100,Error,a1,I,LVLEDUC,Missingness,LVLEDUC must be present,LVLEDUC cannot be blank,Q10b. LVLVEDUC (level of education) cannot be blank,IF EDUCLVL=blank,,,Yes,No,
+a1-ivp-c-101,101,Error,a1,I,LVLEDUC,Conformity,LVLEDUC conforms,"LVLEDUC must be between 1 and 6, or 9",Q10b. LVLVEDUC (level of education) must be between 1 and 6 or 9,IF LVLEDUC<1 or (EDUC>6 and  is not 9),,,Yes,No,
+a1-ivp-m-102,102,Error,a1,I,MARISTAT,Missingness,MARISTAT must be present,MARISTAT cannot be blank,Q11. MARISTAT (marital status) cannot be blank,IF MARISTAT=blank,,,Yes,Yes,
+a1-ivp-c-103,103,,a1,I,MARISTAT,Conformity,MARISTAT conforms,MARISTAT must be between 1 and 6 or 9,"Q11. MARISTAT (marital status) must be between 1 and 6, or 9",IF MARISTAT<1 or (MARISTAT>6 and is not 9),,,Yes,Yes,
```
Fixing in the file for re-upload!

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-m-112,112,Error,a1,I,MEDVA,Missingness,MEDVA conditionally blank,If SERVED is 0 then MEDVA must be blank,If Q15. SERVED (served in armed forces) is 0 then Q16. MEDVA (VA medical care) must be blank,IF SERVED=0 and MEDVA is not blank,,,Yes,No,
+a1-ivp-c-113,113,Error,a1,I,MEDVA,Conformity,MEDVA conforms,"If SERVED is 1 then MEDVA must be 0, 1, or 9","If Q15. SERVED (served in armed forces) is 1 then Q16. MEDVA (VA medical care) must be 0, 1 or 9","IF SERVED=1 and (MEDVA is not 0,1,9)",,,Yes,No,
+a1-ivp-m-114,114,Error,a1,I,EXRTIME,Missingness,EXRTIME must be present,EXRTIME cannot be blank,Q17. EXRTIME (excercise time) cannot be blank,IF EXRTIME=blank,,,Yes,No,
+a1-ivp-c-115,115,,a1,I,EXRTIME,Conformity,EXRTIME conforms,"EXRTIME must be between 1 and 4, 8, or 9","Q17. EXRTIME (excercise time) must be between 1 and 4, 8, or 9","IF EXRTIME<1 or (EXRTIME>4 and  is not 8,9)",,,Yes,No,
```
Fixing in the file for re-upload!

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-m-120,120,Error,a1,I,MEMTEN,Missingness,MEMTEN must be present,MEMTEN cannot be blank,Q20. MEMTEN (memory past ten years) cannot be blank,IF MEMTEN=blank,,,Yes,No,
+a1-ivp-c-121,121,Error,a1,I,MEMTEN,Conformity,MEMTEN conforms,"MEMTEN must be between 1 and 5, or 9","Q20. MEMTEN (memory past ten years) must be between 1 and 5, or 9",IF MEMTEN<1 or (MEMTEN>5 and MEMTEN is not 9),,,Yes,No,
+a1-ivp-m-122,122,Error,a1,I,ADISTATE,Missingness,ADISTATE conforms,ADISTATE must be between 1 and 10,Q21. ADISTATE (ADI state-decile) must be between 1 and 10,IF ADISTATE <1 or ADISTATE >10,,,Yes,No,
+a1-ivp-m-123,123,Error,a1,I,ADINAT,Missingness,ADINAT conforms,ADINAT must be between 1 and 100,Q22. ADINAT (ADI national percentile) must be between 1 and 100,IF ADINAT <1 or ADINAT >100,,,Yes,No,
```
Yes, fixing in the file for re-upload!

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-c-110,110,Error,a1,I,SERVED,Conformity,SERVED conforms,"SERVED must be 0, 1, or 9","Q15. SERVED (served in armed forces) must be 0, 1, or 9","IF SERVED is not 0,1,9",,,Yes,No,
+a1-ivp-m-111,111,Error,a1,I,MEDVA,Missingness,MEDVA conditionally present,If SERVED is 1 then MEDVA cannot be blank,If Q15. SERVED (served in armed forces) is 1 then Q16. MEDVA (VA medical care) cannot be blank,IF SERVED=1 and MEDVA=blank,,,Yes,No,
+a1-ivp-m-112,112,Error,a1,I,MEDVA,Missingness,MEDVA conditionally blank,If SERVED is 0 then MEDVA must be blank,If Q15. SERVED (served in armed forces) is 0 then Q16. MEDVA (VA medical care) must be blank,IF SERVED=0 and MEDVA is not blank,,,Yes,No,
+a1-ivp-c-113,113,Error,a1,I,MEDVA,Conformity,MEDVA conforms,"If SERVED is 1 then MEDVA must be 0, 1, or 9","If Q15. SERVED (served in armed forces) is 1 then Q16. MEDVA (VA medical care) must be 0, 1 or 9","IF SERVED=1 and (MEDVA is not 0,1,9)",,,Yes,No,
```
hmm I think they can still technically answer MEDVA since they may not remember whether they served in the military but it may be clear through medical records or more recent memory that they had VA care. Aileen said maybe we should write an alert for when MEDVA= 1 (Yes) and SERVED=9 (Unknown) because SERVED should really be equal to 1 in that case. but it may happen so it's better as an alert vs an
 error. I added this as a new edited version of error a1-ivp-p-1006

kgauthreaux commented on forms/uds/a1/form_a1_ivp_error_checks_mc.csv:
```diff
@@ -0,0 +1,152 @@
+a1-ivp-c-004,004,Error,a1,I,BIRTHYR,Conformity,BIRTHYR conforms,BIRTHYR must be between 1850 and current year minus 15,Q1b. BIRTHYR (birth year) must be between 1850 and current year minus 15,IF BIRTHYR<1850 or BIRTHYR>current year minus 15,,,Yes,Yes,
+a1-ivp-m-005,005,Error,a1,I,CHLDHDCTRY,Missingness,CHLDHDCTRY must be present,CHLDHDCTRY cannot be blank,Q2. CHLDHDCTRY (childhood country or region) cannot be blank,IF CHLDHDCTRY=blank,,,Yes,No,
+a1-ivp-c-006,006,Error,a1,I,RACEAIAN,Conformity,RACEAIAN conforms,RACEAIAN must be blank or 1,Q3a. RACEAIAN (race native american) must be blank or 1,IF RACEAIAN is not blank or RACEAIAN  is not 1,,,Yes,No,
+a1-ivp-m-007,007,Error,a1,I,RACEAIANX,Missingness,RACEAIANX conditionally blank,If RACEAIAN is blank then RACEAIANX must be blank,If Q3a. RACEAIAN (race native american) is blank then Q3a1. RACEAIANX (race native american specify) must be blank,IF RACEAIAN is blank and RACEAIANX is not blank,,,Yes,No,
```
these can be blank, all of the subcategories are optional, it's only required that at least one of the "main" categories is selected (RACEAIAN, RACEASIAN, RACEBLACK, ETHISPANIC, RACEMENA, RACENHPI, RACEWHITE, RACEUNKN)

kgauthreaux commented:
@hewanadun hmm, I did upload the latest version of the a1 questions and vars yesterday but yeah I think I changed from 'CSV' to 'UTF-8 CSV' upon Zach's request for the DED. Should I re-upload as the previous format to get this to merge and then create a completely new pull request to get it back to the 'UTF-8 CSV' format?

hewanadun reviewed (approved):
File names for questions_and_vars files in main branch and this branch were different. I deleted the old file and uploaded the one from main branch.

echeng06 commented:
@kgauthreaux I know this branch was closed so you can open a new one, but we need to update the following:

* For `a1-ivp-c-113` and `a1-ivp-c-133`, the compatibility is handled by a different rule so we should we update the CSV to remove it for consistency, e.g.
    * `If SERVED is 1 then MEDVA must be 0,1,9` -> `MEDVA must be 0,1,9` 
    * `If REFERSC is 1,2 then REFLEARNED must be 1-10 or 88,90` -> `REFLEARNED must be 1-10 or 88,90`
* `a1-ivp-m-076` for `GENMAN` needs `GENOTHX` removed from `comp_vars`
